doc/on_chain_spec.md

```diff
@@ -669,3 +669,175 @@ that distributing their respective portions of the auction fee
+  -- ^ Which price did the bidder set to buy the auction lot?
+  , sellerSignature :: BuiltinByteString
+  -- ^ New field. Represents the signed payload
+  }
+
+data AuctionTerms = AuctionTerms
+  { auctionLot :: AssetClass
+  , sellerPKH :: PubKeyHash
+  -- ^ Renamed from seller to sellerPKH
+  , sellerVK :: BuiltinByteString
+  -- ^ New field. Seller verification key
+  ...
+  -- Other fields unchanged
+  }
+```
+
+
+```haskell
+validNewBid
+  :: AuctionTerms
+  -> StandingBidState
+  -> StandingBidState
+  -> Bool
+validNewBid AuctionTerms{..} oldStandingBidState newStandingBidState =
+  isJust $ do
+    let StandingBidState oldApprovedBidders oldBid = oldStandingBidState
+    let StandingBidState newApprovedBidders newBid = newStandingBidState
+    BidTerms bidderPKH newPrice sellerSignature <- newBid
+    -- The seller has allowed the bidder to participate in the auction
+    guard $ verifyEd25519Signature (sellerVK terms) (sellerSignatureMessage (hydraHeadId terms) bidderPKH) sellerSignature
+    -- The new bid respects the starting price and minimum bid increment
+    -- conditions in the auction terms.
+    guard $ case oldBid of
+      Nothing ->
+        startingPrice <= newPrice
+      Just (BidTerms _ oldPrice) ->
+        oldPrice + minimumBidIncrement <= newPrice
+    -- The new standing bid state maintains the same list of approved bidders.
+    guard $ oldApprovedBidders == newApprovedBidders
+```
+
+
+## Signed bidder datums
+
+This section describes an extension to the above spec,
+such that the standing bid is extended with a signed payload by the bidder.
+
+It assumes the `Open Auctions` extension to be implemented
+as it relies on the seller
+for enstablishing the on-chain link between bidder verification key and pubkey hash
+
+
+The idea is to verify on-chain
+that the bidder has signed their new bid,
+and is meant to prevent cases where
+a delegate could submit a false bid for an unaware bidder.
+
+To verify something on-chain,
+we need access to the verification key that produced the signature.
+We already have the bidder `PubKeyHash` available in the `BidTerm`,
+but that can not be used to verify a signature produced by the bidders signing key.
+Unfortunately, it is not possible to even derive the `PubKeyHash` from the verification key on-chain,
+so we will have to carry around both in the datum.
+
+To do this, we must modify `BidTerms` with the bidder vkey,
+and the signature, representing the signed payload.
+
+```haskell
+data BidTerms = BidTerms
+  { bidderPKH :: PubKeyHash
+  -- ^ PubKeyHash of whoever submitted the bid
+  , bidderVK :: BuiltinByteString
+  -- ^ New field. Verification Key of whoever submitted the bid
+  , bidPrice :: Natural
+  -- ^ Which price did the bidder set to buy the auction lot?
+  , bidderSignature :: BuiltinByteString
+  -- ^ New field. Represents the signed payload by the bidder
+  , sellerSignature :: BuiltinByteString
+  -- ^ Represents the signed payload by the seller
+  }
```

Comment:
Needs to mention the auction's voucher asset class. The bidder is bidding in a specific auction, not just announcing a number to the heavens. 😄

Reply:
I am using the `headId` as unique identifier of the auction

Reply:
Ahhh makes sense, thanks!